File: src/main/java/mffs/TransferMode.java
package mffs;

/**
 * The force field transfer mode.
 */
public enum TransferMode
{
	EQUALIZE, DISTRIBUTE, DRAIN, FILL;

	public TransferMode toggle()
	{
		int newOrdinal = this.ordinal() + 1;

		if (newOrdinal >= TransferMode.values().length)
		{
			newOrdinal = 0;
		}
		return TransferMode.values()[newOrdinal];
	}

}

File: src/main/scala/mffs/fortron/TransferMode.scala
package mffs.fortron

/**
 * The force field transfer mode.
 */
object TransferMode extends Enumeration
{
  type TransferMode = Value
  val EQUALIZE, DISTRIBUTE, DRAIN, FILL = Value

  implicit class TransferModeValue(value: Value)
  {
    def toggle: TransferMode = TransferMode.values((value.id + 1) % TransferMode.values.size)
  }

}